client/src/pages/Payments.js

```diff
@@ -0,0 +1,69 @@
+import React, { useState } from "react";
+import { Elements, StripeProvider } from 'react-stripe-elements';
+import InjectedCreditCard from "./CreditCard";
+import ChargePayment from "./ChargePayment";
+import Transfer from "./Transfer";
+import Refund from "./Refund";
+
+var Payments = function () {
+    let userID = 1;
+    const [stripePage, setStripePage] = useState(null);
+    const [clientSecret, setClientSecret] = useState(null);
+
+    if (!clientSecret) {
+        (async () => {
+            const response = await fetch(`/users/${userID}/payments/secret`);
+            const { client_secret: newClientSecret } = await response.json();
+            setClientSecret(newClientSecret);
+        })();
+    }
+
+    function changeStripePage(event) {
+        setStripePage(event.target.value);
+    }
+
+    const optionsMenu = (
+        <div>
+            <select onChange={changeStripePage}>
+                <option value="none">---</option>
+                <option value="creditCard">Add or update my credit card information</option>
+                <option value="chargePayment">Make a payment</option>
+                <option value="transfer">Set up my bank account to receive transfers</option>
+                <option value="refund">Claim a refund</option>
+            </select>
+        </div >
+    )
+
+    const creditCardPage = (
+        <div>
+            <div>
+                <StripeProvider apiKey="pk_test_3Ty6VUy1rfVdHm4JSOP1Uo8z00w8r5ooyx">
+                    <Elements>
+                        <InjectedCreditCard key={1} clientSecret={clientSecret} userID={userID} setupOrUpdate={"setup"} />
+                    </Elements>
+                </StripeProvider>
+            </div>
+            <div>
+                <StripeProvider apiKey="pk_test_3Ty6VUy1rfVdHm4JSOP1Uo8z00w8r5ooyx">
+                    <Elements>
+                        <InjectedCreditCard key={2} clientSecret={clientSecret} userID={userID} setupOrUpdate={"update"} />
+                    </Elements>
+                </StripeProvider>
+            </div>
+        </div>
+    );
+
+    if (stripePage === 'creditCard') {
+        return [optionsMenu, creditCardPage];
+    } else if (stripePage === 'transfer') {
+        return [optionsMenu, <Transfer userID={userID} />];
+    } else if (stripePage === 'chargePayment') {
+        return [optionsMenu, <ChargePayment userID={userID} />];
+    } else if (stripePage === 'refund') {
+        return [optionsMenu, <Refund userID={userID} />];
+    } else {
+        return optionsMenu;
```

Comment:
better to write it like: 

```
render (
  {optionsMenu} # because this is constant in every option
  {stripePage === 'creditCard' && ({creditCardPage})
  {stripePage === 'transfer' && (<Transfer userId={userId}>)
  etc etc
)
```

Reply:
this is a bit more React-y way of writing it :)